File: php/src/Model/WorkerLocationResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * A worker's location after a mobile update.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class WorkerLocationResult
{
    public function __construct(
        public readonly string $workerId,
        public readonly float $latitude,
        public readonly float $longitude,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            workerId: (string) ($data['workerId'] ?? ''),
            latitude: (float) ($data['latitude'] ?? 0),
            longitude: (float) ($data['longitude'] ?? 0),
        );
    }
}

File: java/src/main/java/io/fivexer/sdk/model/WorkerLocationResult.java
package io.fivexer.sdk.model;

/**
 * A worker's location after a mobile update.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class WorkerLocationResult {
    private String workerId;
    private Double latitude;
    private Double longitude;

    public String getWorkerId() { return workerId; }
    public Double getLatitude() { return latitude; }
    public Double getLongitude() { return longitude; }
}
